source/create.ts

```diff
@@ -16,52 +51,55 @@ const aliases = [
 	'delete'
 ];
 
-const create = defaults => {
-	defaults = merge({}, defaults);
-	preNormalizeArguments(defaults.options);
+const create = (defaults: Partial<Defaults>): Got => {
+	defaults = merge<Defaults, Partial<Defaults>>({}, defaults);
+	preNormalizeArguments(defaults.options!);
 
 	if (!defaults.handler) {
 		// This can't be getPromiseOrStream, because when merging
 		// the chain would stop at this point and no further handlers would be called.
 		defaults.handler = (options, next) => next(options);
 	}
 
-	function got(url, options?: any) {
+	// @ts-ignore Because the for loop handles it for us, as well as the other Object.defines
+	const got: Got = (url: URL | string | Partial<Options>, options?: Partial<Options>): ProxyStream | CancelableRequest<Response> => {
 		try {
-			return defaults.handler(normalizeArguments(url, options, defaults), getPromiseOrStream);
+			return defaults.handler!(normalizeArguments(url, options as NormalizedOptions, defaults), getPromiseOrStream);
 		} catch (error) {
 			if (options && options.stream) {
 				throw error;
 			} else {
+				// @ts-ignore It's an Error not a response, but TS thinks it's calling .resolve
```

Comment:
Is there an issue about this on the TypeScript repo?

Reply:
No.. I'm a bit busy so I cannot write it right now, if you want to do it feel free to! ❤️

Reply:
Me too right now, so let's keep that comment here :)